osquery/tables/system/darwin/system_info.cpp

```diff
@@ -0,0 +1,81 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <osquery/database.h>
+#include <osquery/tables.h>
+#include "osquery/core/conversions.h"
+
+#include <CoreFoundation/CoreFoundation.h>
+#include <IOKit/IOKitLib.h>
+
+namespace osquery {
+namespace tables {
+
+Status getCpuId(std::string& cpu_id) {
+  auto matching = IOServiceMatching("IOPlatformExpertDevice");
```

Comment:
If you make the cpu_id variable a local static you can copy that on every additional call to getCpuId. Since the CPU serial won't change!